src/background/stores/card-selection-store.ts

```diff
@@ -79,11 +81,11 @@ export class CardSelectionStore extends PersistentStore<CardSelectionStoreData>
     };
 
     private toggleRuleExpandCollapse = (payload: RuleExpandCollapsePayload): void => {
-        if (!payload || !this.state.rules[payload.ruleId]) {
+        if (!payload || !this.state.rules || !this.state.rules[payload.ruleId]) {
```

Comment:
nit: You can shorten this (and similar checks throughout the file) with `?.` syntax:

```suggestion
        if (!payload || !this.state.rules?.[payload.ruleId]) {
```